### PR TITLE
Renamed STS

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Closes https://github.com/weaveworks/eksctl/issues/4885.

The final part is to just rename STSV2 to STS.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/eks/api.go [hunk 1]
```diff
@@ -300,7 +299,7 @@ func (c *ClusterProvider) GetCredentialsEnv() ([]string, error) {
 
 // checkAuth checks the AWS authentication
 func (c *ClusterProvider) checkAuth() error {
-	output, err := c.Provider.STSV2().GetCallerIdentity(context.TODO(), &stsv2.GetCallerIdentityInput{})
+	output, err := c.Provider.STS().GetCallerIdentity(context.TODO(), &sts.GetCallerIdentityInput{})
```
Comment (cPu1) [inline pkg/eks/api.go:302]: We should propagate the context here.

**Review by cPu1 — APPROVED** [review]
Just one comment about context propagation but otherwise LGTM.

File: pkg/eks/api.go (hunk 1 above)
Comment (Skarlso) [reply pkg/eks/api.go:302]: From where?

Comment (cPu1) [reply pkg/eks/api.go:302]: I meant by adding a `Context` parameter to `checkAuth` and passing it from the top-most call.

Comment (Skarlso) [reply pkg/eks/api.go:302]: I don't necessarily agree with that. :) But that's not a hill I'll die on. :D

Comment (aclevername) [reply pkg/eks/api.go:302]: I'd agree with @cPu1 , we've done that for other v2 migrations, not sure why we wouldn't here :thinking:

**Review by aclevername — APPROVED** [review]
lgtm- 1 comment

File: pkg/eks/api.go (hunk 1 above)
Comment (Skarlso) [reply pkg/eks/api.go:302]: Fine, like I said, I don't agree, but I don't care enough. :D

Comment (Skarlso) [reply pkg/eks/api.go:302]: you are the ones who will review it hahahaha

Comment (Skarlso) [reply pkg/eks/api.go]: Done :)

File: pkg/ctl/deregister/cluster.go [hunk 2]
```diff
@@ -1,12 +1,14 @@
 package deregister
 
 import (
+	context2 "context"
```
Comment (Skarlso) [inline pkg/ctl/deregister/cluster.go:4]: Huh, weird refactor action Goland... I'll fix this in a sec.

File: pkg/eks/api.go (hunk 1 above)
Comment (Skarlso) [reply pkg/eks/api.go]: Okay, it's not that bad actually. :D

File: pkg/actions/cluster/get.go [hunk 3]
```diff
@@ -58,7 +59,7 @@ func GetClusters(provider api.ClusterProvider, listAllRegions bool, chunkSize in
 			continue
 		}
 		// Reset region and recreate the client.
-		ctl, err := newClusterProvider(&api.ProviderConfig{
+		ctl, err := newClusterProvider(context.TODO(), &api.ProviderConfig{
```
Comment (aclevername) [inline pkg/actions/cluster/get.go:62]: shouldn't this be passed in here too?

Comment (Skarlso) [reply pkg/actions/cluster/get.go:62]: It's a serious question. :D Like, is it `GetClusters`? Or is it `getAndPrinterClusters` or is it `doGetCluster` which is added in 

```go
	cmd.CobraCommand.RunE = func(_ *cobra.Command, args []string) error {
		cmd.NameArg = cmdutils.GetNameArg(args)
		return doGetCluster(cmd, params, listAllRegions)
	}
```

That's added from `getClusterCmd` that's added in `Command` which is initalised in main. :)

Comment (aclevername) [reply pkg/actions/cluster/get.go:62]: I'd stop at ctl

Comment (Skarlso) [reply pkg/actions/cluster/get.go]: Ok, moved it out until ctl.

**Pull request merged**